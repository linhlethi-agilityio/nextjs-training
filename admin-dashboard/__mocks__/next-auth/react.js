export const useSession = () => ({
  data: {
    user: {
      token: 'access-token',
      email: 'user@gmail.com',
      id: '1',
      name: 'User',
      role: 'admin',
    },
    expires: Date.now().toString(),
  },
  status: 'authenticated',
  update: async () => null,
});
