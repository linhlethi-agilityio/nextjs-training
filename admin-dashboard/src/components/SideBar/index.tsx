'use client';

import { Box, VStack } from '@chakra-ui/react';
import { usePathname } from 'next/navigation';

// Constants
import { MENU } from '@/constants';

// Components
import { SideBarItem } from '@/components';

const SideBar = () => {
  const pathname = usePathname();

  return (
    <Box w={144} px={4} h="100vh">
      <VStack mt={172}>
        {MENU.map(({ label, icon, path }) => (
          <SideBarItem
            isFocused={path === pathname}
            label={label}
            icon={icon}
            path={path}
            key={`side-${label}`}
          />
        ))}
      </VStack>
    </Box>
  );
};

export default SideBar;
