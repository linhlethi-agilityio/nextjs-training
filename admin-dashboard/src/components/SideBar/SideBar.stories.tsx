import type { Meta, StoryObj } from '@storybook/react';

// Components
import { SideBar } from '@/components';

const meta: Meta<typeof SideBar> = {
  component: SideBar,
};

export default meta;
type Story = StoryObj<typeof SideBar>;

export const Default: Story = {
  args: {},
};
