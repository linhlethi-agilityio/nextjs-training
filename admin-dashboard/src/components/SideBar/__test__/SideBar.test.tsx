import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import SideBar from '..';

describe('SideBar component', () => {
  it('should match snapshot for SideBar', () => {
    const { container } = render(<SideBar />);

    expect(container).toMatchSnapshot();
  });
});
