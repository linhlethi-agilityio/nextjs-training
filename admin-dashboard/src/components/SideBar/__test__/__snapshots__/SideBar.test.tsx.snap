// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component should match snapshot for SideBar 1`] = `
<div>
  <div
    class="css-76c7c3"
  >
    <div
      class="chakra-stack css-awxekx"
    >
      <a
        href="/"
      >
        <div
          class="chakra-stack css-1s41oml"
          data-testid="sidebar-item-Dashboard"
        >
          <div
            class="css-k008qs"
          >
            <svg
              class="chakra-icon css-z2vag5"
              focusable="false"
              viewBox="0 0 13 13"
            >
              <g
                clip-path="url(#clip0_16_90)"
              >
                <path
                  d="M4.60461 1.5V7.5H1.60461V1.5H4.60461ZM1.60461 0.75C1.4057 0.75 1.21494 0.829018 1.07428 0.96967C0.933632 1.11032 0.854614 1.30109 0.854614 1.5L0.854614 7.5C0.854614 7.69891 0.933632 7.88968 1.07428 8.03033C1.21494 8.17098 1.4057 8.25 1.60461 8.25H4.60461C4.80353 8.25 4.99429 8.17098 5.13494 8.03033C5.2756 7.88968 5.35461 7.69891 5.35461 7.5V1.5C5.35461 1.30109 5.2756 1.11032 5.13494 0.96967C4.99429 0.829018 4.80353 0.75 4.60461 0.75H1.60461ZM11.3546 2.25V6H7.60461V2.25H11.3546ZM7.60461 1.5C7.4057 1.5 7.21494 1.57902 7.07428 1.71967C6.93363 1.86032 6.85461 2.05109 6.85461 2.25V6C6.85461 6.19891 6.93363 6.38968 7.07428 6.53033C7.21494 6.67098 7.4057 6.75 7.60461 6.75H11.3546C11.5535 6.75 11.7443 6.67098 11.8849 6.53033C12.0256 6.38968 12.1046 6.19891 12.1046 6V2.25C12.1046 2.05109 12.0256 1.86032 11.8849 1.71967C11.7443 1.57902 11.5535 1.5 11.3546 1.5H7.60461ZM4.60461 10.5V12H3.10461V10.5H4.60461ZM3.10461 9.75C2.9057 9.75 2.71494 9.82902 2.57428 9.96967C2.43363 10.1103 2.35461 10.3011 2.35461 10.5V12C2.35461 12.1989 2.43363 12.3897 2.57428 12.5303C2.71494 12.671 2.9057 12.75 3.10461 12.75H4.60461C4.80353 12.75 4.99429 12.671 5.13494 12.5303C5.2756 12.3897 5.35461 12.1989 5.35461 12V10.5C5.35461 10.3011 5.2756 10.1103 5.13494 9.96967C4.99429 9.82902 4.80353 9.75 4.60461 9.75H3.10461ZM12.1046 9V10.5H7.60461V9H12.1046ZM7.60461 8.25C7.4057 8.25 7.21494 8.32902 7.07428 8.46967C6.93363 8.61032 6.85461 8.80109 6.85461 9V10.5C6.85461 10.6989 6.93363 10.8897 7.07428 11.0303C7.21494 11.171 7.4057 11.25 7.60461 11.25H12.1046C12.3035 11.25 12.4943 11.171 12.6349 11.0303C12.7756 10.8897 12.8546 10.6989 12.8546 10.5V9C12.8546 8.80109 12.7756 8.61032 12.6349 8.46967C12.4943 8.32902 12.3035 8.25 12.1046 8.25H7.60461Z"
                  fill="currentColor"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_16_90"
                >
                  <rect
                    fill="white"
                    height="12"
                    transform="translate(0.854614 0.75)"
                    width="12"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="chakra-text css-d4sxp6"
          >
            Dashboard
          </p>
        </div>
      </a>
      <a
        href="/product"
      >
        <div
          class="chakra-stack css-1s41oml"
          data-testid="sidebar-item-Product"
        >
          <div
            class="css-k008qs"
          >
            <svg
              class="chakra-icon css-z2vag5"
              focusable="false"
              viewBox="0 0 10 12"
            >
              <path
                d="M8.33057 0H2.33057C1.93274 0 1.55121 0.158035 1.26991 0.43934C0.988602 0.720644 0.830566 1.10218 0.830566 1.5V10.5C0.830566 10.8978 0.988602 11.2794 1.26991 11.5607C1.55121 11.842 1.93274 12 2.33057 12H8.33057C8.72839 12 9.10992 11.842 9.39123 11.5607C9.67253 11.2794 9.83057 10.8978 9.83057 10.5V1.5C9.83057 1.10218 9.67253 0.720644 9.39123 0.43934C9.10992 0.158035 8.72839 0 8.33057 0ZM3.08057 3H7.58057C7.68002 3 7.77541 3.03951 7.84573 3.10984C7.91606 3.18016 7.95557 3.27554 7.95557 3.375C7.95557 3.47446 7.91606 3.56984 7.84573 3.64016C7.77541 3.71049 7.68002 3.75 7.58057 3.75H3.08057C2.98111 3.75 2.88573 3.71049 2.8154 3.64016C2.74508 3.56984 2.70557 3.47446 2.70557 3.375C2.70557 3.27554 2.74508 3.18016 2.8154 3.10984C2.88573 3.03951 2.98111 3 3.08057 3ZM2.70557 4.875C2.70557 4.77554 2.74508 4.68016 2.8154 4.60984C2.88573 4.53951 2.98111 4.5 3.08057 4.5H7.58057C7.68002 4.5 7.77541 4.53951 7.84573 4.60984C7.91606 4.68016 7.95557 4.77554 7.95557 4.875C7.95557 4.97446 7.91606 5.06984 7.84573 5.14016C7.77541 5.21049 7.68002 5.25 7.58057 5.25H3.08057C2.98111 5.25 2.88573 5.21049 2.8154 5.14016C2.74508 5.06984 2.70557 4.97446 2.70557 4.875ZM3.08057 6H7.58057C7.68002 6 7.77541 6.03951 7.84573 6.10983C7.91606 6.18016 7.95557 6.27554 7.95557 6.375C7.95557 6.47446 7.91606 6.56984 7.84573 6.64017C7.77541 6.71049 7.68002 6.75 7.58057 6.75H3.08057C2.98111 6.75 2.88573 6.71049 2.8154 6.64017C2.74508 6.56984 2.70557 6.47446 2.70557 6.375C2.70557 6.27554 2.74508 6.18016 2.8154 6.10983C2.88573 6.03951 2.98111 6 3.08057 6ZM3.08057 7.5H5.33057C5.43002 7.5 5.52541 7.53951 5.59573 7.60983C5.66606 7.68016 5.70557 7.77554 5.70557 7.875C5.70557 7.97446 5.66606 8.06984 5.59573 8.14017C5.52541 8.21049 5.43002 8.25 5.33057 8.25H3.08057C2.98111 8.25 2.88573 8.21049 2.8154 8.14017C2.74508 8.06984 2.70557 7.97446 2.70557 7.875C2.70557 7.77554 2.74508 7.68016 2.8154 7.60983C2.88573 7.53951 2.98111 7.5 3.08057 7.5Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <p
            class="chakra-text css-d4sxp6"
          >
            Product
          </p>
        </div>
      </a>
      <a
        href="/category"
      >
        <div
          class="chakra-stack css-1s41oml"
          data-testid="sidebar-item-Category"
        >
          <div
            class="css-k008qs"
          >
            <svg
              class="chakra-icon css-z2vag5"
              focusable="false"
              viewBox="0 0 13 13"
            >
              <g
                clip-path="url(#clip0_16_96)"
              >
                <path
                  d="M10.2056 2.5C10.305 2.5 10.4004 2.53951 10.4707 2.60984C10.5411 2.68016 10.5806 2.77554 10.5806 2.875C10.5806 2.97446 10.5411 3.06984 10.4707 3.14016C10.4004 3.21049 10.305 3.25 10.2056 3.25H6.45557C6.35611 3.25 6.26073 3.21049 6.1904 3.14016C6.12008 3.06984 6.08057 2.97446 6.08057 2.875C6.08057 2.77554 6.12008 2.68016 6.1904 2.60984C6.26073 2.53951 6.35611 2.5 6.45557 2.5H10.2056ZM10.2056 4.75C10.305 4.75 10.4004 4.78951 10.4707 4.85984C10.5411 4.93016 10.5806 5.02554 10.5806 5.125C10.5806 5.22446 10.5411 5.31984 10.4707 5.39016C10.4004 5.46049 10.305 5.5 10.2056 5.5H6.45557C6.35611 5.5 6.26073 5.46049 6.1904 5.39016C6.12008 5.31984 6.08057 5.22446 6.08057 5.125C6.08057 5.02554 6.12008 4.93016 6.1904 4.85984C6.26073 4.78951 6.35611 4.75 6.45557 4.75H10.2056ZM10.5806 7.375C10.5806 7.27554 10.5411 7.18016 10.4707 7.10983C10.4004 7.03951 10.305 7 10.2056 7H6.45557C6.35611 7 6.26073 7.03951 6.1904 7.10983C6.12008 7.18016 6.08057 7.27554 6.08057 7.375C6.08057 7.47446 6.12008 7.56984 6.1904 7.64017C6.26073 7.71049 6.35611 7.75 6.45557 7.75H10.2056C10.305 7.75 10.4004 7.71049 10.4707 7.64017C10.5411 7.56984 10.5806 7.47446 10.5806 7.375ZM10.2056 9.25C10.305 9.25 10.4004 9.28951 10.4707 9.35983C10.5411 9.43016 10.5806 9.52554 10.5806 9.625C10.5806 9.72446 10.5411 9.81984 10.4707 9.89017C10.4004 9.96049 10.305 10 10.2056 10H6.45557C6.35611 10 6.26073 9.96049 6.1904 9.89017C6.12008 9.81984 6.08057 9.72446 6.08057 9.625C6.08057 9.52554 6.12008 9.43016 6.1904 9.35983C6.26073 9.28951 6.35611 9.25 6.45557 9.25H10.2056Z"
                  fill="currentColor"
                />
                <path
                  d="M12.8306 1.75C12.8306 1.35218 12.6725 0.970644 12.3912 0.68934C12.1099 0.408035 11.7284 0.25 11.3306 0.25L2.33057 0.25C1.93274 0.25 1.55121 0.408035 1.26991 0.68934C0.988602 0.970644 0.830566 1.35218 0.830566 1.75L0.830566 10.75C0.830566 11.1478 0.988602 11.5294 1.26991 11.8107C1.55121 12.092 1.93274 12.25 2.33057 12.25H11.3306C11.7284 12.25 12.1099 12.092 12.3912 11.8107C12.6725 11.5294 12.8306 11.1478 12.8306 10.75V1.75ZM3.83057 1V11.5H2.33057C2.13165 11.5 1.94089 11.421 1.80024 11.2803C1.65958 11.1397 1.58057 10.9489 1.58057 10.75V1.75C1.58057 1.55109 1.65958 1.36032 1.80024 1.21967C1.94089 1.07902 2.13165 1 2.33057 1H3.83057ZM4.58057 1H11.3306C11.5295 1 11.7202 1.07902 11.8609 1.21967C12.0015 1.36032 12.0806 1.55109 12.0806 1.75V10.75C12.0806 10.9489 12.0015 11.1397 11.8609 11.2803C11.7202 11.421 11.5295 11.5 11.3306 11.5H4.58057V1Z"
                  fill="currentColor"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_16_96"
                >
                  <rect
                    fill="white"
                    height="12"
                    transform="translate(0.830566 0.25)"
                    width="12"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="chakra-text css-d4sxp6"
          >
            Category
          </p>
        </div>
      </a>
      <a
        href="/agen"
      >
        <div
          class="chakra-stack css-1s41oml"
          data-testid="sidebar-item-Agen"
        >
          <div
            class="css-k008qs"
          >
            <svg
              class="chakra-icon css-z2vag5"
              focusable="false"
              viewBox="0 0 17 17"
            >
              <g
                clip-path="url(#clip0_16_65)"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.93057 11.7C3.97714 11.6379 4.03753 11.5875 4.10696 11.5528C4.17639 11.5181 4.25294 11.5 4.33057 11.5H6.83057C6.96317 11.5 7.09035 11.5527 7.18412 11.6464C7.27789 11.7402 7.33057 11.8674 7.33057 12C7.33057 12.1326 7.27789 12.2598 7.18412 12.3536C7.09035 12.4473 6.96317 12.5 6.83057 12.5H4.58057L2.33057 15.5H15.3306L13.0806 12.5H10.8306C10.698 12.5 10.5708 12.4473 10.477 12.3536C10.3832 12.2598 10.3306 12.1326 10.3306 12C10.3306 11.8674 10.3832 11.7402 10.477 11.6464C10.5708 11.5527 10.698 11.5 10.8306 11.5H13.3306C13.4082 11.5 13.4847 11.5181 13.5542 11.5528C13.6236 11.5875 13.684 11.6379 13.7306 11.7L16.7306 15.7C16.7863 15.7743 16.8202 15.8626 16.8285 15.9551C16.8369 16.0476 16.8193 16.1406 16.7778 16.2236C16.7363 16.3067 16.6724 16.3765 16.5934 16.4253C16.5144 16.4741 16.4234 16.5 16.3306 16.5H1.33057C1.23771 16.5 1.14669 16.4741 1.0677 16.4253C0.988713 16.3765 0.92488 16.3067 0.883353 16.2236C0.841827 16.1406 0.824248 16.0476 0.832587 15.9551C0.840926 15.8626 0.874853 15.7743 0.930567 15.7L3.93057 11.7Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M8.83057 1.5C8.4366 1.5 8.04649 1.5776 7.68252 1.72836C7.31854 1.87913 6.98782 2.10011 6.70925 2.37868C6.43067 2.65726 6.20969 2.98798 6.05893 3.35195C5.90816 3.71593 5.83057 4.10604 5.83057 4.5C5.83057 4.89397 5.90816 5.28408 6.05893 5.64805C6.20969 6.01203 6.43067 6.34275 6.70925 6.62132C6.98782 6.8999 7.31854 7.12088 7.68252 7.27164C8.04649 7.42241 8.4366 7.5 8.83057 7.5C9.62622 7.5 10.3893 7.18393 10.9519 6.62132C11.5145 6.05871 11.8306 5.29565 11.8306 4.5C11.8306 3.70435 11.5145 2.94129 10.9519 2.37868C10.3893 1.81607 9.62622 1.5 8.83057 1.5ZM4.83057 4.5C4.83064 3.73023 5.05282 2.97683 5.47046 2.33019C5.8881 1.68356 6.48345 1.17117 7.18507 0.85451C7.88669 0.537845 8.66479 0.430355 9.42599 0.54494C10.1872 0.659524 10.8992 0.991315 11.4765 1.5005C12.0538 2.00968 12.4719 2.67463 12.6807 3.41556C12.8894 4.15648 12.88 4.94191 12.6534 5.67759C12.4269 6.41327 11.9929 7.06796 11.4035 7.5631C10.8141 8.05823 10.0943 8.37278 9.33057 8.469V14C9.33057 14.1326 9.27789 14.2598 9.18412 14.3536C9.09035 14.4473 8.96318 14.5 8.83057 14.5C8.69796 14.5 8.57078 14.4473 8.47701 14.3536C8.38325 14.2598 8.33057 14.1326 8.33057 14V8.47C7.36353 8.34817 6.47426 7.87744 5.82978 7.14624C5.18531 6.41504 4.82999 5.47368 4.83057 4.499V4.5Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_16_65"
                >
                  <rect
                    fill="white"
                    height="16"
                    transform="translate(0.830566 0.5)"
                    width="16"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="chakra-text css-d4sxp6"
          >
            Agen
          </p>
        </div>
      </a>
      <a
        href="/analytic"
      >
        <div
          class="chakra-stack css-1s41oml"
          data-testid="sidebar-item-Analytic"
        >
          <div
            class="css-k008qs"
          >
            <svg
              class="chakra-icon css-z2vag5"
              focusable="false"
              viewBox="0 0 17 17"
            >
              <path
                d="M4.83057 11.75H2.83057V14.75H4.83057V11.75ZM9.83057 7.75H7.83057V14.75H9.83057V7.75ZM14.8306 2.75V14.75H12.8306V2.75H14.8306ZM12.8306 1.75C12.5653 1.75 12.311 1.85536 12.1235 2.04289C11.9359 2.23043 11.8306 2.48478 11.8306 2.75V14.75C11.8306 15.0152 11.9359 15.2696 12.1235 15.4571C12.311 15.6446 12.5653 15.75 12.8306 15.75H14.8306C15.0958 15.75 15.3501 15.6446 15.5377 15.4571C15.7252 15.2696 15.8306 15.0152 15.8306 14.75V2.75C15.8306 2.48478 15.7252 2.23043 15.5377 2.04289C15.3501 1.85536 15.0958 1.75 14.8306 1.75H12.8306ZM6.83057 7.75C6.83057 7.48478 6.93592 7.23043 7.12346 7.04289C7.311 6.85536 7.56535 6.75 7.83057 6.75H9.83057C10.0958 6.75 10.3501 6.85536 10.5377 7.04289C10.7252 7.23043 10.8306 7.48478 10.8306 7.75V14.75C10.8306 15.0152 10.7252 15.2696 10.5377 15.4571C10.3501 15.6446 10.0958 15.75 9.83057 15.75H7.83057C7.56535 15.75 7.311 15.6446 7.12346 15.4571C6.93592 15.2696 6.83057 15.0152 6.83057 14.75V7.75ZM1.83057 11.75C1.83057 11.4848 1.93592 11.2304 2.12346 11.0429C2.311 10.8554 2.56535 10.75 2.83057 10.75H4.83057C5.09578 10.75 5.35014 10.8554 5.53767 11.0429C5.72521 11.2304 5.83057 11.4848 5.83057 11.75V14.75C5.83057 15.0152 5.72521 15.2696 5.53767 15.4571C5.35014 15.6446 5.09578 15.75 4.83057 15.75H2.83057C2.56535 15.75 2.311 15.6446 2.12346 15.4571C1.93592 15.2696 1.83057 15.0152 1.83057 14.75V11.75Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <p
            class="chakra-text css-d4sxp6"
          >
            Analytic
          </p>
        </div>
      </a>
      <a
        href="/setting"
      >
        <div
          class="chakra-stack css-1s41oml"
          data-testid="sidebar-item-Setting"
        >
          <div
            class="css-k008qs"
          >
            <svg
              class="chakra-icon css-z2vag5"
              focusable="false"
              viewBox="0 0 17 17"
            >
              <g
                clip-path="url(#clip0_16_72)"
              >
                <path
                  d="M8.8306 5.504C8.40432 5.504 7.98223 5.58796 7.5884 5.75109C7.19458 5.91421 6.83675 6.15331 6.53533 6.45473C6.23391 6.75615 5.99481 7.11399 5.83168 7.50781C5.66856 7.90163 5.58459 8.32373 5.58459 8.75C5.58459 9.17627 5.66856 9.59837 5.83168 9.99219C5.99481 10.386 6.23391 10.7438 6.53533 11.0453C6.83675 11.3467 7.19458 11.5858 7.5884 11.7489C7.98223 11.912 8.40432 11.996 8.8306 11.996C9.69149 11.996 10.5171 11.654 11.1259 11.0453C11.7346 10.4365 12.0766 9.61089 12.0766 8.75C12.0766 7.88911 11.7346 7.06347 11.1259 6.45473C10.5171 5.84599 9.69149 5.504 8.8306 5.504ZM6.5846 8.75C6.5846 8.15432 6.82123 7.58304 7.24243 7.16184C7.66364 6.74063 8.23492 6.504 8.8306 6.504C9.42627 6.504 9.99755 6.74063 10.4188 7.16184C10.84 7.58304 11.0766 8.15432 11.0766 8.75C11.0766 9.34568 10.84 9.91695 10.4188 10.3382C9.99755 10.7594 9.42627 10.996 8.8306 10.996C8.23492 10.996 7.66364 10.7594 7.24243 10.3382C6.82123 9.91695 6.5846 9.34568 6.5846 8.75Z"
                  fill="currentColor"
                />
                <path
                  d="M10.6266 2.093C10.0996 0.303 7.56155 0.303 7.03455 2.093L6.94055 2.412C6.90375 2.53695 6.83945 2.65208 6.75237 2.74896C6.66529 2.84583 6.55763 2.92199 6.4373 2.97185C6.31696 3.02171 6.18699 3.04401 6.05691 3.03712C5.92683 3.03022 5.79995 2.99431 5.68555 2.932L5.39355 2.772C3.75355 1.88 1.96055 3.674 2.85355 5.313L3.01255 5.605C3.07486 5.71939 3.11078 5.84628 3.11767 5.97635C3.12457 6.10643 3.10227 6.2364 3.05241 6.35674C3.00255 6.47708 2.92639 6.58474 2.82951 6.67182C2.73264 6.7589 2.61751 6.8232 2.49255 6.86L2.17355 6.954C0.383555 7.481 0.383555 10.019 2.17355 10.546L2.49255 10.64C2.61751 10.6768 2.73264 10.7411 2.82951 10.8282C2.92639 10.9153 3.00255 11.0229 3.05241 11.1433C3.10227 11.2636 3.12457 11.3936 3.11767 11.5236C3.11078 11.6537 3.07486 11.7806 3.01255 11.895L2.85255 12.187C1.96055 13.827 3.75355 15.621 5.39355 14.727L5.68555 14.568C5.79995 14.5057 5.92683 14.4698 6.05691 14.4629C6.18699 14.456 6.31696 14.4783 6.4373 14.5281C6.55763 14.578 6.66529 14.6542 6.75237 14.751C6.83945 14.8479 6.90375 14.963 6.94055 15.088L7.03455 15.407C7.56155 17.197 10.0996 17.197 10.6266 15.407L10.7206 15.088C10.7574 14.963 10.8217 14.8479 10.9087 14.751C10.9958 14.6542 11.1035 14.578 11.2238 14.5281C11.3442 14.4783 11.4741 14.456 11.6042 14.4629C11.7343 14.4698 11.8612 14.5057 11.9756 14.568L12.2676 14.728C13.9076 15.621 15.7016 13.826 14.8076 12.187L14.6486 11.895C14.5862 11.7806 14.5503 11.6537 14.5434 11.5236C14.5365 11.3936 14.5588 11.2636 14.6087 11.1433C14.6586 11.0229 14.7347 10.9153 14.8316 10.8282C14.9285 10.7411 15.0436 10.6768 15.1686 10.64L15.4876 10.546C17.2776 10.019 17.2776 7.481 15.4876 6.954L15.1686 6.86C15.0436 6.8232 14.9285 6.7589 14.8316 6.67182C14.7347 6.58474 14.6586 6.47708 14.6087 6.35674C14.5588 6.2364 14.5365 6.10643 14.5434 5.97635C14.5503 5.84628 14.5862 5.71939 14.6486 5.605L14.8086 5.313C15.7016 3.673 13.9066 1.88 12.2676 2.773L11.9756 2.932C11.8612 2.99431 11.7343 3.03022 11.6042 3.03712C11.4741 3.04401 11.3442 3.02171 11.2238 2.97185C11.1035 2.92199 10.9958 2.84583 10.9087 2.74896C10.8217 2.65208 10.7574 2.53695 10.7206 2.412L10.6266 2.093ZM7.99355 2.376C8.23955 1.541 9.42155 1.541 9.66755 2.376L9.76155 2.695C9.84058 2.96309 9.97859 3.21009 10.1655 3.4179C10.3524 3.62571 10.5834 3.78906 10.8416 3.89598C11.0999 4.0029 11.3787 4.05067 11.6578 4.03579C11.9369 4.02091 12.2092 3.94376 12.4546 3.81L12.7456 3.65C13.5096 3.235 14.3456 4.07 13.9296 4.835L13.7706 5.127C13.637 5.37243 13.56 5.64463 13.5453 5.92366C13.5306 6.20268 13.5785 6.48147 13.6855 6.73958C13.7925 6.9977 13.9559 7.2286 14.1637 7.41538C14.3715 7.60215 14.6185 7.74006 14.8866 7.819L15.2046 7.913C16.0396 8.159 16.0396 9.341 15.2046 9.587L14.8856 9.681C14.6175 9.76002 14.3705 9.89804 14.1627 10.0849C13.9548 10.2718 13.7915 10.5028 13.6846 10.7611C13.5777 11.0193 13.5299 11.2982 13.5448 11.5773C13.5596 11.8564 13.6368 12.1286 13.7706 12.374L13.9306 12.665C14.3456 13.429 13.5106 14.265 12.7456 13.849L12.4546 13.69C12.2091 13.5563 11.9368 13.4791 11.6576 13.4643C11.3785 13.4495 11.0996 13.4974 10.8413 13.6044C10.5831 13.7114 10.3521 13.8749 10.1652 14.0828C9.97839 14.2907 9.84046 14.5378 9.76155 14.806L9.66755 15.124C9.42155 15.959 8.23955 15.959 7.99355 15.124L7.89955 14.805C7.8205 14.5371 7.68251 14.2902 7.49569 14.0825C7.30886 13.8748 7.07795 13.7115 6.81984 13.6046C6.56174 13.4977 6.283 13.4499 6.00402 13.4646C5.72505 13.4794 5.45292 13.5564 5.20755 13.69L4.91555 13.85C4.15155 14.265 3.31555 13.43 3.73155 12.665L3.89055 12.374C4.02451 12.1286 4.10181 11.8562 4.11678 11.577C4.13176 11.2978 4.08403 11.0188 3.9771 10.7604C3.87017 10.5021 3.70675 10.2709 3.49883 10.084C3.29092 9.89702 3.04378 9.75899 2.77555 9.68L2.45655 9.586C1.62155 9.34 1.62155 8.158 2.45655 7.912L2.77555 7.818C3.04336 7.73891 3.29009 7.60094 3.4977 7.41419C3.7053 7.22744 3.86852 6.99663 3.97541 6.73866C4.0823 6.48068 4.13015 6.20207 4.11547 5.92322C4.10079 5.64437 4.02394 5.37232 3.89055 5.127L3.73055 4.835C3.31555 4.071 4.15055 3.235 4.91555 3.651L5.20755 3.81C5.45292 3.94358 5.72505 4.02057 6.00402 4.03535C6.283 4.05013 6.56174 4.00232 6.81984 3.89542C7.07795 3.78851 7.30886 3.62523 7.49569 3.41752C7.68251 3.20981 7.8205 2.96295 7.89955 2.695L7.99355 2.376Z"
                  fill="currentColor"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_16_72"
                >
                  <rect
                    fill="white"
                    height="16"
                    transform="translate(0.830566 0.75)"
                    width="16"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="chakra-text css-d4sxp6"
          >
            Setting
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
